src/pages/SetPasswordPage.js

```diff
@@ -69,14 +70,26 @@ class SetPasswordPage extends Component {
      * Validate the form and then submit it
      */
     validateAndSubmitForm() {
+        const accountID = lodashGet(this.props.route.params, 'accountID', '');
+        const validateCode = lodashGet(this.props.route.params, 'validateCode', '');
         if (!this.state.isFormValid) {
             return;
         }
-        setPassword(
-            this.state.password,
-            lodashGet(this.props.route, 'params.validateCode', ''),
-            lodashGet(this.props.route, 'params.accountID', ''),
-        );
+        API.ValidateEmail({
+            accountID,
+            validateCode,
+        }).then((responseValidate) => {
+            if (responseValidate.jsonCode === 200) {
```

Comment:
No error message if this fails?

Reply:
And if the `ValidateEmail` call fails.. what is the user supposed to do? 

Another thought, if the validate code is bad why make them enter a password at all?

Reply:
- Hmmm. Ask for another code, I imagine?
- Mainly because we don't know 😛. What are you thinking? Calling validate on page load, for example?

Reply:
>Ask for another code, I imagine?

Ahaha asking because when I ran through this flow I just get stuck looking at an error message with no buttons or anything.

>Calling validate on page load, for example? That would probably work.

Sure, this is what we do on OldDot.

Reply:
We can create another issue to nail these two down, if you want. I think validating on load makes sense and would make the flow a bit better, and if validation fails, we can display the error and a button/link to resend the magic link?

I don't think we need them to go live with N6, but they seem like good improvements